packages/firestore/src/core/sync_engine.ts

```diff
@@ -450,6 +452,7 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
       (this.isPrimary && source === OnlineStateSource.RemoteStore) ||
       (!this.isPrimary && source === OnlineStateSource.SharedClientState)
     ) {
+      this.assertSubscribed('applyOnlineStateChange()');
```

Comment:
Is this intended?

Reply:
Yes. This was missing but should be present. Otherwise `this.syncEngineListener!.*` might hit errors.